### PR TITLE
Refactor logic for parallel testing

### PR DESCRIPTION
**Pull request by gerashegalov**
Reduce code duplication in run_pyspark_from_build.sh. 

Borrowing logic tested in #1688 

Tested with
```
# single threaded
TEST_PARALLEL=1 SPARK_HOME=~/spark-3.1.1-bin-hadoop3.2 integration_tests/run_pyspark_from_build.sh -k test_single_orderby

# parallel: 
TEST_PARALLEL=4 SPARK_HOME=~/spark-3.1.1-bin-hadoop3.2 integration_tests/run_pyspark_from_build.sh -k test_single_orderby
```

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: integration_tests/run_pyspark_from_build.sh [hunk 1]
```diff
@@ -99,41 +99,29 @@ else
     ## Under cloud environment, overwrite the '--std_input_path' param to point to the distributed file path
     INPUT_PATH=${INPUT_PATH:-"$SCRIPTPATH"}
 
+    RUN_TESTS_COMMAND="$SCRIPTPATH"/runtests.py\ --rootdir\ "$LOCAL_ROOTDIR"\ "$LOCAL_ROOTDIR"/src/main/python
```
Comment (jlowe) [inline integration_tests/run_pyspark_from_build.sh:102]: I think this is equivalent and simpler to read.
```suggestion
    RUN_TESTS_COMMAND="$SCRIPTPATH/runtests.py --rootdir $LOCAL_ROOTDIR $LOCAL_ROOTDIR/src/main/python"
```

However with this PR's refactoring there is another layer of shell parsing.  I believe that causes issues when any of the path variables expand to paths with spaces in them (or other shell-sensitive characters when re-parsed).

Comment (gerashegalov) [reply integration_tests/run_pyspark_from_build.sh:102]: yes, it's quite likely it does not for edge cases. I presumed it does not apply to our test infra. If necessary I will look for a solution, and open for suggestions. 

re: suggested removal of the double-quotes I agree, I decided to keep it from existing code because it's not breaking anything afaik.

Comment (jlowe) [reply integration_tests/run_pyspark_from_build.sh:102]: >  I presumed it does not apply to our test infra.

This script is not specific to our infra.  It can run in any environment with the requisite Python modules and has `SPARK_HOME` setup.  Any user pulling the repo into a parent directory with a space in it will not be able to run this script after this change but could before it.

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Comment by pxLi** [comment]
I saw this already been triggered not sure why trigger twice for the same commit. If the previous one is not required plz help abort to save resource, thx~

[rapids_premerge-github 1253](https://blossom.nvidia.com/sw-gpu-spark-jenkins/job/rapids_premerge-github/1253/)
[rapids_premerge-github 1254](https://blossom.nvidia.com/sw-gpu-spark-jenkins/job/rapids_premerge-github/1254/)

**Review by jlowe — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

**Comment by gerashegalov** [comment]
@pxLi sorry, my browser UI probably did not refresh and I mistakenly thought my first build comment didn't go through. Now I see that I have duplicate build comments.
